### ISSUE TITLE
Sunsetting the beta channel

### ISSUE THREAD

**Issue by lidel** [ipfs-shipyard/ipfs-companion#964]
#  TLDR: switch to the Stable channel ASAP

| <img src="https://unpkg.com/@browser-logos/firefox/firefox_16x16.png" width="16" height="16"> [Firefox](https://www.mozilla.org/firefox/new/) / [Firefox for Android](https://play.google.com/store/apps/details?id=org.mozilla.firefox) | <img src="https://unpkg.com/@browser-logos/chrome/chrome_16x16.png" width="16" height="16"> [Chrome](https://www.google.com/chrome/) / <img src="https://unpkg.com/@browser-logos/brave/brave_16x16.png" width="16" height="16"> [Brave](https://brave.com/) / <img src="https://unpkg.com/@browser-logos/opera/opera_16x16.png" width="16" height="16"> [Opera](https://www.opera.com/)  / <img src="https://unpkg.com/@browser-logos/edge/edge_16x16.png" width="16" height="16"> [Edge](https://www.microsoftedgeinsider.com/)
|------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| [![Install From AMO](https://ipfs.io/ipfs/QmWNa64XjA78QvK3zG2593bSMizkDXXcubDHjnRDYUivqt)](https://addons.mozilla.org/firefox/addon/ipfs-companion/) | [![Install from Chrome Store](https://ipfs.io/ipfs/QmXeTTMAxJVSZLqNcVzBdsAZKhWUpP7w7QAZ8f3Bnmk3Mj)](https://chrome.google.com/webstore/detail/ipfs-companion/nibjojkomfdiaoajekhjakgkdhaomnch)  |

# Rationale

v2.17.1 illustrated problems with Chrome Web Store when publishing bugfix releases. 
Below is a plan to improve current state of things.

## Current setup

We have a dedicated Beta channel (which has separate ID) and we published a fix release there first.  
After a day, we decided the fix is important enough that we also submitted it to the Stable channel. 

Stable channel got reviewed within 1-2h, while Beta was still pending review, which defeats its purpose as a platform for fast-tracked testing. Based on our observations during 2020, the number of users impacts the priority in the review queue (~40k stable vs ~200 beta). With the user count on the Stable channel, reviews do not seem to ne an issue, and we should leverage that.

## Proposed change

Deprecate Beta channel.
Chrome Web Store supports staged rollouts, where published can define % of users to get autoupdate.

This enables us to have a single publishing channel (current Stable) and initially publish to 1% of userbase (~400 atm), then do all the QA we need. It also enables us to bump it to 10%, 25% 50% 75% each following day and monitor github/forums, increasing our confidence and making big changes much less stressful.

### Added value

- Collapsing a lot of complexity and red tape
- Single channel could be automated with less effort (https://github.com/ipfs-shipyard/ipfs-companion/issues/778)
- Improved UX when enthusiasts want to test new release (no need to disable Stable and switch to Beta, no need to manually migrate settings)

### TODOs

- [ ] we should stop announcing it in release notes, readmes and docs
- [ ] we should add code that detects user is running Beta version and urging them to switch to stable channel
  - could be big red message in the main menu, in the place of https://github.com/ipfs-shipyard/ipfs-companion/pull/935
  - I strongly believe the version should remain functional, but there should be permanent call for action urgin user to switch to the stable channel. 
  - we might be able to signal when stable/beta is installed #beta and disable beta when stable is present (if there is no API, we may leverage `ipfs block get|rm`)
-  [ ] keep Beta channel unlisted "just in case" (errors/mistakes happen, extension ID for Beta is safelisted in Brave so it may be handy to temporarily switch to different ID if we want to quickly fix some upstream issue with Chrome Web Store)

**Comment by jessicaschilling** [comment]
@lidel This sounds great. I like the idea of using the "new version alert" position from #935 but replacing it with an alert triangle (`glyph_attention`) in `.red` (#ea5037).

We've embedded those icons as SVG, so https://gateway.ipfs.io/ipfs/QmQYzfzGKAeBWgarrmTSsG2PJ8AFfFEBprhVBFaudcFTof/icons/glyph_attention.svg would be the place to grab that code from, but I'm happy to provide anything else if you need it 😊
